src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1804,6 +1804,41 @@ public void TestLongProcessIsWorking()
             Assert.True(p.HasExited);
         }
 
+        [Fact]
+        public void GetProcesses_LongProcessName()
+        {
+            string commandName = "sleep";
+
+            // sleep program doesn't exist on some flavor
+            if (!IsProgramInstalled(commandName))
+            {
+                return;
+            }
+
+            string longProcessName = "123456789012345678901234567890";
+            string sleepCommandPathFileName = Path.Combine(TestDirectory, longProcessName);
+            File.Copy(GetProgramPath(commandName), sleepCommandPathFileName);
+
+            // start sleep program and wait for some seconds
+            using (Process px = Process.Start(new ProcessStartInfo { FileName = sleepCommandPathFileName , Arguments = "30", UseShellExecute = true}))
+            {
+                bool processFound = false;
+                foreach (Process process in Process.GetProcesses())
+                {
+                    if (process.ProcessName == longProcessName)
+                    {
+                        processFound = true;
+                        break;
+                    }
+                }
+                px.Kill();
+                px.WaitForExit();
```

Comment:
It'd be better to have these, in particular the Kill, in a finally block, so that if some unexpected error happens, we don't leave a process running on the machine.  We can also utilize Assert.Contains to simplify this whole thing, e.g.
```C#
try
{
    Assert.Contains(Process.GetProcesses(), p => p.ProcessName == longProcessName);
}
finally
{
    px.Kill();
    px.WaitForExit();
}

Reply:
ok